cmd/chart-repo/utils_test.go

```diff
@@ -285,19 +285,62 @@ func Test_parseRepoIndex(t *testing.T) {
 func Test_chartsFromIndex(t *testing.T) {
 	r := repo{Name: "test", URL: "http://testrepo.com"}
 	index, _ := parseRepoIndex([]byte(validRepoIndexYAML))
-	charts := chartsFromIndex(index, r)
+	charts := chartsFromIndex(index, r, new(filters))
 	assert.Equal(t, len(charts), 2, "number of charts")
-
 	indexWithDeprecated := validRepoIndexYAML + `
   deprecated-chart:
   - name: deprecated-chart
     deprecated: true`
 	index2, err := parseRepoIndex([]byte(indexWithDeprecated))
 	assert.NoErr(t, err)
-	charts = chartsFromIndex(index2, r)
+	charts = chartsFromIndex(index2, r, new(filters))
 	assert.Equal(t, len(charts), 2, "number of charts")
 }
 
+func Test_chartsFromIndexFilterByName(t *testing.T) {
+	r := repo{Name: "test", URL: "http://testrepo.com"}
+	index, _ := parseRepoIndex([]byte(validRepoIndexYAML))
+	filter := new(filters)
+	filter.Names = append(filter.Names, "wordpress")
+	filter.Names = append(filter.Names, "not-found")
+	charts := chartsFromIndex(index, r, filter)
+	assert.Equal(t, len(charts), 1, "number of charts")
+}
+
+func Test_chartsFromIndexFilterByAnnotationWithValue(t *testing.T) {
+	r := repo{Name: "test", URL: "http://testrepo.com"}
+	index, _ := parseRepoIndex([]byte(validRepoIndexYAML))
+	filter := new(filters)
+	filter.Annotations = make(map[string]string)
+	filter.Annotations["sync"] = "true"
+	filter.Annotations["not-found"] = "missing"
```

Comment:
Is the expectation here that it only needs to match one of the annotations and not all of the annotations? I would have assumed that all the annotations need to match for the entry to be selected (I believe this is also how `kubectl` label filters work).

Reply:
Good point I did not think about that.  My intend was if any of them matched it synced.  That is actually how I just rewrote the code as a function.

```
// filterEntry return true if entry matches a filter attribute
func filterEntry( entry *helmrepo.ChartVersion, filter *filters) bool 
````